### ISSUE TITLE
PackageDeclaration: false positive "Missing package declaration" on module-info.java

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#21110]
I have read check documentation: https://checkstyle.org/checks/coding/packagedeclaration.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

Note: module-info.java is only parseable after #8240, so this is reproduced on that branch.

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="PackageDeclaration"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > mod/module-info.java <<'EOF'
module com.example.app {
    requires java.base;

    exports com.example.api;
}
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.10.0-SNAPSHOT-all.jar -c config.xml mod/module-info.java

Starting audit...
[ERROR] mod/module-info.java:1:1: Missing package declaration. [PackageDeclaration]
Audit done.
Checkstyle ends with 1 errors.
```

**Describe what you expect in detail.**

No violation. A modular compilation unit cannot declare a package, so "Missing package declaration" can never be satisfied on module-info.java.

[JLS 7.3](https://docs.oracle.com/javase/specs/jls/se25/html/jls-7.html#jls-7.3) defines `ModularCompilationUnit` as `{ImportDeclaration} ModuleDeclaration`, with no package declaration, so adding one is a compile error rather than a fix for this violation.

The Check already handles the analogous case for compact source files. `beginTree` records whether the root is a compact compilation unit and `finishTree` skips the violation for it:

```java
@Override
public void beginTree(DetailAST ast) {
    defined = false;
    isCompactSourceFile = ast != null
            && ast.getType() == TokenTypes.COMPACT_COMPILATION_UNIT;
}

@Override
public void finishTree(DetailAST ast) {
    if (!defined && !isCompactSourceFile && ast != null) {
        log(ast, MSG_KEY_MISSING);
    }
}
```

A modular compilation unit needs the same treatment. Its root is `COMPILATION_UNIT` with a `MODULE_DEF` child, so it is detectable in `beginTree` the same way.

Expected on the input above: no violations.

**Comment by romani** [comment]
ok, we need to fix this in same release as new grammar to let user not notice it at all.
@vivek-0509 , please create Pr with two commits  and make fix ready, when we will start merging we will just drop first commit in PR for this bug fix. No retesting will be required and we will merge quickly.

**Comment by vivek-0509** [comment]
@romani we have one more check that has false positive on module-info.java files other than that we do have any false positives.. opening the issue for it as well and will fix it in the same release...

**Comment by romani** [comment]
Fix is merged
